pkg/datapath/linux/probes/probes_privileged_test.go

```diff
@@ -48,3 +50,26 @@ func (s *ProbesPrivTestSuite) TestHelpers(c *C) {
 	_, ok := pm.GetHelpers("sched_act")["bpf_map_lookup_elem"]
 	c.Assert(ok, Equals, true)
 }
+
+func (s *ProbesPrivTestSuite) TestWriteHeaders(c *C) {
+	var buf bytes.Buffer
+
+	expectedPatterns := []string{
+		"#ifndef BPF_FEATURES_H_",
+		"#define BPF_FEATURES_H_",
+		"#define HAVE_BPF_SYSCALL",
+		"#define HAVE.*PROG_TYPE",
+		"#define HAVE.*MAP_TYPE",
+		"#define HAVE.*HELPER",
+		"#endif /* BPF_FEATURES_H_ */",
+	}
+
+	writer := bufio.NewWriter(&buf)
+	pm := NewProbeManager()
+	pm.writeHeaders(writer)
+	content := buf.String()
+
+	for _, pattern := range expectedPatterns {
+		c.Assert(content, Matches, pattern)
```

Comment:
Unfortunately, gocheck's `Matches` [requires regular expressions that match the full string](https://github.com/adjust/gocheck/blob/fbc315b36e0e7dc95023e96b43af5d8de58bc6fb/checkers.go#L307). A first solution would be to frame `pattern` with match-all statements:
```go
c.Assert(content, Matches, "(.|\\n)*"+pattern+"(.|\\n)*")
```
But that's very ugly. A better solution is probably to define our own `PartialMatches` in `pkg/checker/checker.go` to allow matching against a subset of the given string. That would look something like that:
```go
type matchesChecker struct {
	*check.CheckerInfo
}

// PartialMatches is a GoCheck checker that the provided regex matches at least
// part of the provided string. It can act as a substitute for Matches.
var (
	matchesParams                = []string{"value", "regex"}
	PartialMatches check.Checker = &matchesChecker{
		&check.CheckerInfo{Name: "PartialMatches", Params: matchesParams},
	}
)

func (checker *matchesChecker) Check(params []interface{}, _ []string) (result bool, error string) {
	return matches(params[0], params[1])
}

func matches(value, regex interface{}) (result bool, error string) {
	regexStr, ok := regex.(string)
	if !ok {
		return false, "Regex must be a string"
	}
	valueStr, ok := value.(string)
	if !ok {
		return false, "Value must be a string"
	}
	matches, err := regexp.MatchString(regexStr, valueStr)
	if err != nil {
		return false, "Failed to compile regex: " + err.Error()
	}
	return matches, ""
}
```